components/web_discovery/browser/hash_detection.cc

```diff
@@ -0,0 +1,72 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/web_discovery/browser/hash_detection.h"
+
+#include <cmath>
+
+#include "base/containers/fixed_flat_map.h"
+#include "brave/components/web_discovery/browser/hash_detection_matrix.h"
+
+namespace web_discovery {
+
+namespace {
+
+constexpr auto kTokenMap = base::MakeFixedFlatMap<char, size_t>(
```

Comment:
not sure how often do we call `IsHashLikely`, but anyway i would write slightly more optimal and readable code. i.e. a function like
```
size_t ctoi(char c) {
  if (c >= 'a') return c - 'a' + 10;
  if (c >= 'A') return c - 'A' + 36;
  return c - '0';
}
```

Reply:
hah, i just copied this from the original implementation not realizing that all the token ids are sequential. will do